cluster-autoscaler/core/scaledown/actuation/actuator.go

```diff
@@ -173,36 +174,53 @@ func (a *Actuator) deleteAsyncEmpty(NodeGroupViews []*budgets.NodeGroupView, nod
 	return reportedSDNodes
 }
 
-// taintNodesSync synchronously taints all provided nodes with NoSchedule. If tainting fails for any of the nodes, already
+// taintNodesAsync asynchronously taints all provided nodes with NoSchedule. If tainting fails for any of the nodes, already
 // applied taints are cleaned up.
-func (a *Actuator) taintNodesSync(NodeGroupViews []*budgets.NodeGroupView) (time.Duration, errors.AutoscalerError) {
+func (a *Actuator) taintNodesAsync(NodeGroupViews []*budgets.NodeGroupView) (time.Duration, errors.AutoscalerError) {
+	failedTaintedNodes := make(map[*apiv1.Node]error)
 	var taintedNodes []*apiv1.Node
 	var updateLatencyTracker *UpdateLatencyTracker
 	nodeDeleteDelayAfterTaint := a.nodeDeleteDelayAfterTaint
 	if a.ctx.AutoscalingOptions.DynamicNodeDeleteDelayAfterTaintEnabled {
 		updateLatencyTracker = NewUpdateLatencyTracker(a.ctx.AutoscalingKubeClients.ListerRegistry.AllNodeLister())
 		go updateLatencyTracker.Start()
 	}
+	nodesLock := sync.Mutex{}
+	var wg sync.WaitGroup
```

Comment:
Using `WaitGroup` directly is ok, though I would consider using a [`ParallelizeUntil`](https://github.com/kubernetes/kubernetes/blob/1504f10e7946f95a8b1da35e28e4c7453ff62775/staging/src/k8s.io/client-go/util/workqueue/parallelizer.go#L46) to simplify the code a bit and allow configurability. While parallelism is nice, perhaps we want to keep # of async calls bound to some constant?

Reply:
Makes sense, I set it to 5 workers